### PR TITLE
Add Usage docs, Integration Tests, refactor export-worker and remove fake S3

### PR DESCRIPTION
**Pull request by lukasmartinelli**
- Add Usage docs

@stirringhalo It is based on your wiki entry. Based on the superbe docs written by you I saw that it is way to complicated. I simplified some components so that users no longer have to use pipecat etc. This happens in the containers now. It is still not perfect, want to extend it with the things you have written in the wiki.

- Integration Tests

Much more complete integration tests now with testing the distributed workflow as well.

- Refactor export-worker

Split the huge Python function and make it stop when no more jobs available. 

- Remove fake S3

I nkow you use this component @stirringhalo in testing.
But I tested with the mock-s3 and it just doesn't work very well - it cannot substitute a real S3 setup (sometimes exceptions happen in the mock-s3 container when replacing files) and especially when rendering it for real this does not really work with the mock-s3.
I now require a real S3 API but it is still configurable so someone can add their own S3 implementation if they want. We use the Switch S3 and not AWS S3 for example in production.

Quite a big PR..not a good example of having concise changes with a single purpose. But need to pump this out before today to deliver it as release.

Oh and I added a CONTRIBUTORS.md file. Please check it out to see whether you are happy what I've written there or edit it.

### CODE REVIEW

File: tools/integration-test/integration_test.py [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+
+    def export_bbox(bbox, min_zoom, max_zoom):
+        dc.run([
+            '-e', 'BBOX={}'.format(bbox),
+            '-e', 'MIN_ZOOM={}'.format(min_zoom),
+            '-e', 'MAX_ZOOM={}'.format(max_zoom),
+            'export'
+        ])
+
+    def verify_no_tiles_missing(x, y, min_z, max_z):
```
Comment (manuelroth) [inline tools/integration-test/integration_test.py]: This function is redundant.

**Comment by hannesj** [comment]
@lukasmartinelli Could you fix my name to `Hannes Junnila`?

**Comment by lukasmartinelli** [comment]
> @lukasmartinelli Could you fix my name to Hannes Junnila?

Fixed in https://github.com/osm2vectortiles/osm2vectortiles/commit/7743a9a94073d1555399c2194c71765ef6c684a0. Sorry

**Comment by stirringhalo** [comment]
> @stirringhalo It is based on your wiki entry. Based on the superbe docs written by you I saw that it is way to complicated. I simplified some components so that users no longer have to use pipecat etc. This happens in the containers now. It is still not perfect, want to extend it with the things you have written in the wiki.

Excellent, been buried with split-polygons and subdivide so I just had a good look at this now. Looks fantastic and really helps! At some point I'll point the wiki at your [USAGE.md](https://raw.githubusercontent.com/osm2vectortiles/osm2vectortiles/master/USAGE.md) document

> I nkow you use this component @stirringhalo in testing. But I tested with the mock-s3 and it just doesn't work very well - it cannot substitute a real S3 setup (sometimes exceptions happen in the mock-s3 container when replacing files) and especially when rendering it for real this does not really work with the mock-s3.

Not a problem, I saw some of the same issues. Like, you can't have much more than 20,000 files in mock-s3 or it gets buried and is very slow. So I stick with zoom-job level 7. I'll keep using it for now as it's handled my previous planet render relatively well. But I'm starting to consider other options down the road.

**Pull request merged**